functions.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/** Query to return all review records for main collection page display
+ *
+ * @param PDO $db
+ *
+ * @return array
+ */
+function dbQueryGetAll(PDO $db) : array {
+    $query = $db->prepare("SELECT `burger_name`, `restaurant`, `visit_date`, `image`, `price`, `patty_rating`, `topping_rating`, `sides_rating`, `value_rating`, `total_score` FROM `reviews`");
+    $query->execute();
+    return $query->fetchAll();
+}
+
+/** Cycles through reviews, changing the date format to dd/mm/yyyy from sql yyyy/mm/dd
+ *
+ * @param array $all_reviews
+ *
+ * @return array
+ */
+function dateFormatUK(array $all_reviews) : array {
+    if ($all_reviews != null) {
+        foreach ($all_reviews as $key => $review) {
+            if (isset($review['visit_date'])) {
+                $date = date_create($review['visit_date']);
+                $all_reviews[$key]['visit_date'] = date_format($date, "d/m/Y");
+            }
+        }
+        return $all_reviews;
+    } else {
+        $output = [];
+        return $output;
+    }
+}
+
+/** Generates html for review panels from db array. Must receive exact fields
+ *
+ * @param array $all_reviews
+ *
+ * @return string
+ */
+function displayReviews(array $all_reviews) : string {
+    $output = '';
+    $required_keys = ['burger_name', 'restaurant', 'visit_date', 'image', 'price', 'patty_rating', 'topping_rating', 'sides_rating', 'value_rating', 'total_score'];
+
+    if ($all_reviews != null) {
+        foreach ($all_reviews as $review) {
+            if (is_array($review) && $review != null) {
+                if (array_keys($review) === $required_keys) {
```

Comment:
good use of array_keys i like this

Reply:
Could this be done in an external function as a check before this function is called?

Reply:
Is this a necessary step? Would it be ok to functionalize it as part of Story 2?

Reply:
would be worth doing sooner rather than later as it will mean that the function is only doing one thing - single responsibility principle